packages/core-flows/src/region/workflows/create-regions.ts

```diff
@@ -1,13 +1,76 @@
-import { CreateRegionDTO, RegionDTO } from "@medusajs/types"
-import { WorkflowData, createWorkflow } from "@medusajs/workflows-sdk"
+import { CreateRegionDTO, WorkflowTypes } from "@medusajs/types"
+import {
+  createWorkflow,
+  transform,
+  WorkflowData,
+} from "@medusajs/workflows-sdk"
 import { createRegionsStep } from "../steps"
-
-type WorkflowInput = { regionsData: CreateRegionDTO[] }
+import {
+  upsertAndReplaceRegionPaymentProvidersStep,
+  UpsertAndReplaceRegionPaymentProvidersStepInput,
+} from "../steps/upsert-and-replace-region-payment-providers"
 
 export const createRegionsWorkflowId = "create-regions"
 export const createRegionsWorkflow = createWorkflow(
   createRegionsWorkflowId,
-  (input: WorkflowData<WorkflowInput>): WorkflowData<RegionDTO[]> => {
-    return createRegionsStep(input.regionsData)
+  (
+    input: WorkflowData<WorkflowTypes.RegionWorkflow.CreateRegionsWorkflowInput>
+  ): WorkflowData<WorkflowTypes.RegionWorkflow.CreateRegionsWorkflowOutput> => {
+    const data = transform(input, (data) => {
+      const regionsData: CreateRegionDTO[] = []
+      const regionsProvidersMap: { [key: string]: string[] } = {}
+
+      for (const region of data.regions) {
+        const { payment_providers, ...rest } = region
+        regionsData.push(rest)
+
+        if (payment_providers?.length) {
+          regionsProvidersMap[JSON.stringify(rest)] = payment_providers
+        }
+      }
+
+      return {
+        regions: regionsData,
+        regions_providers_map: regionsProvidersMap,
+      }
+    })
+
+    const regions = createRegionsStep(data.regions)
+
+    const normalizedRegionProviderData = transform(
```

Comment:
todo: Can we make this easier to read? I think the change of `input` would simplify it a bit.

Can we get rid of the map? You would never update an amount of regions that could cause a bottleneck on performance.

Reply:
I don't think we can simplify, the API allows us to create many regions, each of those regions can have a set of providers, so I am creating a hash for each region before creation which is associated to the set of providers for that region. After the creation, I need to re associate the right region id based on the input hash of the corresponding region to grab the corresponding set of providers. we can discuss it if you have another idea

Reply:
We have quite a few scenarios like this, and we'll have a lot more, in some cases where the hash approach doesn't work.

For the products and prices, I rely on the indices, and I know that is fine because the API guarantees same input/output ordering. What if we set that as a requirement on modules? Then relying on indices would be safe to do, and have the code a lot simpler

Reply:
yes we can go with that approach indeed :thu